client/x/evmstaking/types/withdraw_test.go

```diff
@@ -0,0 +1,202 @@
+package types_test
+
+import (
+	"fmt"
+	"math"
+	"math/big"
+	"testing"
+
+	"github.com/cosmos/cosmos-sdk/types/module/testutil"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/piplabs/story/client/x/evmstaking/types"
+)
+
+type WithdrawTestSuite struct {
+	suite.Suite
+	encConf testutil.TestEncodingConfig
+	delAddr string
+	valAddr string
+	evmAddr common.Address
+}
+
+func (suite *WithdrawTestSuite) SetupTest() {
+	suite.encConf = testutil.MakeTestEncodingConfig()
+	// set dummy addresses
+	suite.delAddr = "story1hmjw3pvkjtndpg8wqppwdn8udd835qpan4hm0y"
+	suite.valAddr = "storyvaloper1hmjw3pvkjtndpg8wqppwdn8udd835qpaa6r6y0"
+	suite.evmAddr = common.HexToAddress("0x131D25EDE18178BAc9275b312001a63C081722d2")
+}
+
+func (suite *WithdrawTestSuite) TestString() {
+	require := suite.Require()
+	// Define the test cases
+	testCases := []struct {
+		name           string
+		withdrawal     types.Withdrawal
+		expectedString string
+	}{
+		{
```

Comment:
Add a test where the resulted string has leading or trailing spaces to test out this[ line](https://github.com/piplabs/story/blob/main/client/x/evmstaking/types/withdraw.go#L24C9-L24C26).

Reply:
@limengformal 
Added at https://github.com/piplabs/story/pull/55/commits/21d7ef85d84c2e4d73b6a153bc5f550889bd4d5d